src/mon/MonMap.cc

```diff
@@ -110,6 +111,26 @@ void mon_info_t::print(ostream& out) const
       << " crush location " << crush_loc;
 }
 
+void mon_info_t::dump(ceph::Formatter *f) const
+{
+  f->dump_string("name", name);
+  f->dump_stream("addr") << public_addrs;
+  f->dump_int("priority", priority);
+  f->dump_float("weight", weight);
+  encode_json("crush_location", crush_loc, f);
```

Comment:
I'm not sure reusing the `encode_json()` is the right thing. The problem is that the `dump()` methods are supposed to format-agnostic – they are getting _abstract_ `Formatter`.

However, the idea to employ a helper function is nice. Maybe `Formatter` already offers a facility for dumping containers?

Reply:
@rzarzynski it will end up almost duplicating the code in ceph_json, i'll need to create a template dump function and then functions for each type. and that will duplicate for all other formats. ceph_json will work for xml and other types (even though ceph dencoder dumping only json)